Fixes of x86_64 compiler

e2e4b6b7 opened the pull request:
- Added name for dune-project (otherwise installation fails)
- Fixed missed escaping for `\r`
- Prohibited indirect assignment (`if ... then a else b fi := 0`) as:
  - It does not work correctly in terms of STI / STA and leads to invalid stack
  - For X86_64 it requires propagation of the assignment as variable may be stored in register
- Added error if assembly toolchain failed (before this compiler returned with 0 code)
- Some minor fixes of correctness

e2e4b6b7 committed:
Fixed \r escaping

e2e4b6b7 committed:
Fixed improper STA usage

e2e4b6b7 committed:
Added fail in case of indirect assignment into register

e2e4b6b7 committed:
Fixed placement of arguments for ".match_failure" call

e2e4b6b7 committed:
Added failing if assembly compiler failed

e2e4b6b7 committed:
Fixed movement of large constants

e2e4b6b7 committed:
Prohibit indirect assignments

Kakadu commented:
@e2e4b6b7 
Could you fix CI?

e2e4b6b7 committed:
Removed ld warning

e2e4b6b7 commented:
Now CI fails in the same way as on the original branch. (There was a branch synchronization issue)

Additionally, I fixed #30 by `-Wa,--noexecstack` argument. But not removed any dirty hacks and not updated tests output as `gen.ml` is not working for me (it results with files with empty output)

Kakadu commented:
> Now CI fails in the same way as on the original branch.

It looks like it started giving an exit code 1. It's scary

![image](https://github.com/user-attachments/assets/5e51c0e5-ff60-439c-8275-90977a737576)

e2e4b6b7 commented:
> It looks like it started giving an exit code 1. It's scary

Yes, but master branch doing the same right now

Kakadu commented:
Could you check that removing redirection and grep helps?

e2e4b6b7 commented:
Yes, looks like it is exactly issue of grep

![image](https://github.com/user-attachments/assets/04a8965c-2172-480d-8d1a-1c28e0b06d07)

Kakadu committed:
[tests] Repair tests in stdlib/regression

I was failing because leftover grep command that
filters out a warning about '.note.GNU-stack'
(No idea where it was fixed)

Signed-off-by: Kakadu <Kakadu@pm.me>

Kakadu commented:
It's better now, but it transformed into a Mac issue

e2e4b6b7 commented:
Also the same issue as on the master. Looks like it tries to build 32-bit runtime on macos
